### PR TITLE
Update domain only flow: `site-or-domain` step

### PR DESCRIPTION
**Pull request by gius80**
## Changes proposed in this Pull Request

This PR enables the reskinned design on domain only flow and updates the second step (`site-or-domain`).

It's part of the new domain-only flow (pcYYhz-ye-p2)

before-logged-in

before-logged-out

after-logged-in

after-logged-out

## Testing instructions

- Build this branch locally or open the live Calypso link
- Navigate to to `/start/domain/domain-only`
- Enter a domain name and select it, in order to move to the next step
- Verify that the layout matches the screenshots above (if you are not logged in the option "Existing site" should be hide)

### CODE REVIEW

File: client/signup/icons/index.tsx [hunk 1]
```diff
@@ -132,3 +132,52 @@ export const upload: ReactElement = (
...
+		/>
+		<path
+			d="M9 21V9"
+			stroke="#8C8F94"
+			strokeWidth="1.6"
+			strokeLinecap="round"
+			strokeLinejoin="round"
+		/>
+		<rect x="3" y="3" width="18" height="18" rx="1" stroke="#8C8F94" strokeWidth="1.6" />
+	</svg>
```
Comment (leonardost) [inline client/signup/icons/index.tsx]: Do these icons need to be added here? I believe they're all standard icons in the `@wordpress/icons` package: `globe`, `addCard` and `layout`

File: client/signup/steps/site-or-domain/index.jsx [hunk 2]
```diff
@@ -2,13 +2,16 @@ import { localize } from 'i18n-calypso';
...
 import { connect } from 'react-redux';
+import HeaderImage from 'calypso/assets/images/domains/domain.svg';
 import QueryProductsList from 'calypso/components/data/query-products-list';
 import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
 import { getDomainProductSlug } from 'calypso/lib/domains';
 import StepWrapper from 'calypso/signup/step-wrapper';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import { getAvailableProductsList } from 'calypso/state/products-list/selectors';
 import { submitSignupStep } from 'calypso/state/signup/progress/actions';
+import { globe, newSite, site } from '../../icons';
```
Comment (leonardost) [inline client/signup/steps/site-or-domain/index.jsx]: As said in the other comment, I believe these could be replaced by `import { globe, addCard, layout } from '@wordpress/icons'`. I experimented doing that:

Screen Shot 2022-01-24 at 19 59 26

I think only the icons' color would need some tweaking 🙂

Comment (leonardost) [reply client/signup/steps/site-or-domain/index.jsx]: I found these by going through the [@wordpress/icons](https://github.com/WordPress/gutenberg/tree/trunk/packages/icons/src/library) source, which wasn't ideal 🤔

I tried to find someplace where we can find icons more easily and after some MGS'ing, found this very helpful link: https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library 😄

**Review by leonardost — APPROVED** [review]
These pages look very nice, thanks for the update! They also work as expected, tested both the logged and unlogged pages. I left just one comment about the icons.

**Comment by gius80** [comment]
Thanks for the review @leonardost! I replaced the icons and also updated the layout according to last design updates.

**Review by leonardost — APPROVED** [review]
Thanks for the updates @gius80, this looks really good! I noticed another small thing, the buttons are not vertically centered with the text:

Markup on 2022-01-27 at 16:25:39

Other than that, LGTM! 🚀

**Comment by gius80** [comment]
Thanks @leonardost, good catch on buttons! I'll fix them.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7120044

Thank you @gius80 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7120044).

**Pull request merged**